awx/ui_next/src/util/useRequest.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState, useRef, useCallback } from 'react';
+
+export default function useRequest(makeRequest, initialValue) {
```

Comment:
![Screenshot from 2020-02-03 09-59-32](https://user-images.githubusercontent.com/9753817/73663621-ef646300-466b-11ea-9cff-eae79ad8220c.png)

You wrote some good details in the PR description - you could and totally should reuse them by putting them into a docstring.